File: task-03/prime.go
package main

import "fmt"

func main() {
    var n int
    fmt.Print("Enter a number: ")
    fmt.Scan(&n)

    L := []int{2}

    for i := 3; i <= n; i++ {
        isPrime := true
        for j := 2; j < i; j++ {
            if i%j == 0 {
                isPrime = false
                break
            }
        }
        if isPrime {
            L = append(L, i)
        }
    }

    if n>=2 {
        for _, L := range L {
            fmt.Print(L, " ")
        }}
}

File: task-03/prime.hs
isPrime :: Int -> Bool
isPrime n
  | n <= 1 = False
  | n == 2 = True
  | otherwise = all (\x -> n `mod` x /= 0) [2..intSqrt n]
  where
    intSqrt = floor . sqrt . fromIntegral

primesUpToN :: Int -> [Int]
primesUpToN n = filter isPrime [2..n]

main :: IO ()
main = do
  putStrLn "Enter a number :"
  input <- getLine
  let n = read input :: Int
  let primes = primesUpToN n
  print primes
